base/src/main/java/com/smartdevicelink/managers/screen/choiceset/BaseChoiceSetManager.java

```diff
@@ -455,16 +459,52 @@ public void setKeyboardConfiguration(@Nullable KeyboardProperties keyboardConfig
         if (keyboardConfiguration == null) {
             this.keyboardConfiguration = defaultKeyboardConfiguration();
         } else {
-            KeyboardProperties properties = new KeyboardProperties();
-            properties.setLanguage((keyboardConfiguration.getLanguage() == null ? Language.EN_US : keyboardConfiguration.getLanguage()));
-            properties.setKeyboardLayout((keyboardConfiguration.getKeyboardLayout() == null ? KeyboardLayout.QWERTZ : keyboardConfiguration.getKeyboardLayout()));
-            properties.setKeypressMode((keyboardConfiguration.getKeypressMode() == null ? KeypressMode.RESEND_CURRENT_ENTRY : keyboardConfiguration.getKeypressMode()));
-            properties.setLimitedCharacterList(keyboardConfiguration.getLimitedCharacterList());
-            properties.setAutoCompleteText(keyboardConfiguration.getAutoCompleteText());
-            this.keyboardConfiguration = properties;
+            this.keyboardConfiguration = createValidKeyboardConfigurationBasedOnKeyboardCapabilitiesFromConfiguration(keyboardConfiguration);
         }
```

Comment:
Because `createValidKeyboardConfigurationBasedOnKeyboardCapabilitiesFromConfiguration` could return `null` we need to avoid setting null to `this.keyboardConfiguration` please consider my suggestion below:
```java
        KeyboardProperties properties = null;
        if (keyboardConfiguration == null) {
            properties = createValidKeyboardConfigurationBasedOnKeyboardCapabilitiesFromConfiguration(keyboardConfiguration);
        }
        if (properties == null) {
            this.keyboardConfiguration = defaultKeyboardConfiguration();
        } else {
            this.keyboardConfiguration = properties;
        }
```

Reply:
Good point. But I think we should always call `createValidKeyboardConfigurationBasedOnKeyboardCapabilitiesFromConfiguration ` even if the passed `keyboardConfiguration` is not null right? do you agree with this suggestion?
```
public void setKeyboardConfiguration(@Nullable KeyboardProperties keyboardConfiguration) {
    KeyboardProperties properties = createValidKeyboardConfigurationBasedOnKeyboardCapabilitiesFromConfiguration(keyboardConfiguration);
    if (properties == null) {
        this.keyboardConfiguration = defaultKeyboardConfiguration();
    } else {
        this.keyboardConfiguration = properties;
    }
}
```

Reply:
@vladmu Thanks for the updates! @bilal-alsharifi I've included this as a separate commit, please, let me know if we should revert it